internal/envoy/v3/route.go

```diff
@@ -122,23 +122,37 @@ func RouteRoute(r *dag.Route) *envoy_route_v3.Route_Route {
 	}
 }
 
-// hashPolicy returns a slice of hash policies iff at least one of the route's
-// clusters supplied uses the `Cookie` load balancing strategy.
-func hashPolicy(r *dag.Route) []*envoy_route_v3.RouteAction_HashPolicy {
-	for _, c := range r.Clusters {
-		if c.LoadBalancerPolicy == "Cookie" {
-			return []*envoy_route_v3.RouteAction_HashPolicy{{
-				PolicySpecifier: &envoy_route_v3.RouteAction_HashPolicy_Cookie_{
-					Cookie: &envoy_route_v3.RouteAction_HashPolicy_Cookie{
-						Name: "X-Contour-Session-Affinity",
-						Ttl:  protobuf.Duration(0),
-						Path: "/",
-					},
+// hashPolicy returns a slice of Envoy hash policies from the passed in Contour
+// request hash policy configuration. Only one of header or cookie hash policies
+// should be set on any RequestHashPolicy element.
+func hashPolicy(r []dag.RequestHashPolicy) []*envoy_route_v3.RouteAction_HashPolicy {
+	if len(r) == 0 {
+		return nil
+	}
+	hashPolicies := []*envoy_route_v3.RouteAction_HashPolicy{}
+	for _, h := range r {
```

Comment:
nit: `h` & `r` aren't super descriptive to understand what each var means.

Reply:
yeah good call, will redo those names